sha1_mb/sha1_mb_mgr_submit_avx2.asm

```diff
@@ -117,63 +117,64 @@ sha1_mb_mgr_submit_avx2:
 	vmovdqa  [rsp + 8*8 + 16*9], xmm15
 %endif
 
-	mov	unused_lanes, [state + _unused_lanes]
-	mov	lane, unused_lanes
-	and	lane, 0xF
+	mov	DWORD(tmp2), _unused_lanes
+	add	tmp2, state 	; tmp2= state + _unused_lanes
+	mov	unused_lanes, [tmp2]
+	mov	DWORD(lane), DWORD(unused_lanes)
+	and	DWORD(lane), 0xF
 	shr	unused_lanes, 4
-	imul	lane_data, lane, _LANE_DATA_size
 	mov	dword [job + _status], STS_BEING_PROCESSED
-	lea	lane_data, [state + _ldata + lane_data]
-	mov	[state + _unused_lanes], unused_lanes
+	lea	lane_data, [state + lane*8]
+	mov	[tmp2], unused_lanes
 	mov	DWORD(len), [job + _len]
 
-	mov	[lane_data + _job_in_lane], job
+	mov	[lane_data + _job_in_lane + _ldata ], job
 
-	shl	len,4
-	or	len, lane
-	mov	[state + _lens + 4*lane], DWORD(len)
+	shl	DWORD(len),4
+	or	DWORD(len), DWORD(lane)
+	lea	p, [state + 4*lane]
+	mov	[p + _lens], DWORD(len)
 	; Load digest words from result_digest
+	; p isn't
 	vmovdqu	xmm0, [job + _result_digest + 0*16]
 	mov	DWORD(tmp), [job + _result_digest + 1*16]
 
-	vmovd   [state + _args_digest + 4*lane + 0*32], xmm0
-	vpextrd [state + _args_digest + 4*lane + 1*32], xmm0, 1
-	vpextrd [state + _args_digest + 4*lane + 2*32], xmm0, 2
-	vpextrd [state + _args_digest + 4*lane + 3*32], xmm0, 3
-	mov     [state + _args_digest + 4*lane + 4*32], DWORD(tmp)
-
+	vmovd   [p + 0*32], xmm0
+	vpextrd [p + 1*32], xmm0, 1
+	vpextrd [p + 2*32], xmm0, 2
+	vpextrd [p + 3*32], xmm0, 3
+	mov     [p + 4*32], DWORD(tmp)
 	mov	p, [job + _buffer]
-	mov	[state + _args_data_ptr + 8*lane], p
+	mov	[lane_data + _args_data_ptr ], p
 
 	add	dword [state + _num_lanes_inuse], 1
-	cmp	unused_lanes, 0xf
+	cmp	DWORD(unused_lanes), 0xf
```

Comment:
Yes but i was puzzled by that because the test will jump to return_null if unused_lanes !=15 so i could not understand why you need a full 64bit register for the numbered of unsued lanes. Anyway i'll switch it back, it's only one byte more :D